packages/core/src/common/message-rpc/rpc-message-encoder.ts

```diff
@@ -348,22 +374,36 @@ export class RpcMessageEncoder {
             }
         });
 
+        this.registerEncoder(ObjectType.Function, {
+            is: value => typeof value === 'function',
+            write: () => { }
+        });
+
         this.registerEncoder(ObjectType.Object, {
             is: value => typeof value === 'object',
-            write: (buf, object, recursiveEncode) => {
-                const properties = Object.keys(object);
-                const relevant = [];
-                for (const property of properties) {
-                    const value = object[property];
-                    if (typeof value !== 'function') {
-                        relevant.push([property, value]);
+            write: (buf, object, visitedObjects, recursiveEncode) => {
+                try {
+                    if (visitedObjects.has(object)) {
+                        throw new EncodingError('Object to encode contains circular references!');
+                    }
+                    visitedObjects.add(object);
+
+                    const properties = Object.keys(object);
+                    const relevant = [];
+                    for (const property of properties) {
+                        const value = object[property];
+                        if (typeof value !== 'function') {
```

Comment:
If this line filters out functions, why do we need an encoder for functions?

Reply:
This line only filters out object properties that are functions. However, we also have to consider that functions can occur as elements of an array. (e.g. 
```ts
[()=> { // do something}, ()=>{// do something else}];
```
)

Of course we could also just filter out functions in the `writeArray` method but I opted for the same approach that JSON.stringify/parse uses and decided to encode this functions as emtpy objects.

I have no hard opinion on this. If you think it's better to filter out functions entirely I'm happy to apply the changes.